example/test_tetanus.py

```diff
@@ -12,71 +14,102 @@
 
 
 class TestsSpecGiraffe:
-    """A Giraffe spec tests base."""
+    """A Giraffe spec tests base.
+
+    Args:
+        giraffe: Giraffe instance.
+    """
+
+    _giraffe: Giraffe = None # type: ignore
+
     @classmethod
-    def setup_class(cls, components):
-        cls._giraffe, *_ = components
+    def setup_class(cls, connections: List[Giraffe]) -> None:
+        """Initializes the variables once in the start of the spec.
+
+        Args:
+            connections: Required components.
+        """
+
+        cls._giraffe, *_ = connections
 
 
 class TestsSpecTetanus(TestsSpecGiraffe):
-    """A Tetanus spec tests."""
+    """A Tetanus spec tests.
+
+    Args:
+        _tetanus_lib: An instance of the tested library.
+        _echo_port: A port to echo on.
+    """
+
+    _tetanus_lib: Tetanus
+    _echo_port: int
 
     @classmethod
     @pytest.mark.lego('giraffe')
-    def setup_class(cls, components):
-        super().setup_class(components)
+    def setup_class(cls, connections: List[Giraffe]) -> None:
+        """Initializes the variables once in the start of the spec.
+
+        Args:
+            connections: Required components.
+        """
 
-        cls._tetanus_lib = example.libs.tetanus.Tetanus()
+        super().setup_class(connections)
+
+        cls._tetanus_lib = Tetanus()
         cls._echo_port = 1337
 
-    def setup_method(self):
-        self._tetanus_lib.install(self._giraffe, TOOL, self._echo_port)
+    def setup_method(self) -> None:
+        """Installes Tetanus at the start of each test."""
+
+        self._tetanus_lib.install(TestsSpecTetanus._giraffe, TOOL, self._echo_port)
+
+    def teardown_method(self) -> None:
+        """Uninstalles Tetanus at the end of each test."""
 
-    def teardown_method(self):
-        self._tetanus_lib.uninstall(self._giraffe)
+        self._tetanus_lib.uninstall(TestsSpecTetanus._giraffe)
 
     @pytest.mark.lego('zebra')
-    async def test_send_and_recv(self, components):
+    async def test_send_and_recv(self, connections): # type: ignore
         "The test send packets and expect them back."""
 
-        zebra, *_ = components
-        await zebra.send_and_receive(self._giraffe.get_ip(), self._echo_port)
+        zebra, *_ = connections
+        await zebra.send_and_receive(TestsSpecTetanus._giraffe.get_ip(), self._echo_port)
 
     @pytest.mark.lego('zebra and elephant')
-    async def test_multi_send_and_recv(self, components):
+    async def test_multi_send_and_recv(self, connections): # type: ignore
         """The test send packets from multiple components and
             expect them back.
         """
 
         tasks = []
 
-        for component in components:
+        for component in connections:
```

Comment:
connection in connections
fix everywhere

Reply:
OK

Reply:
not fixed

Reply:
In my second commit